agent/src/testflinger_agent/client.py

```diff
@@ -220,6 +235,26 @@ def get_result(self, job_id):
         else:
             return {}
 
+    def get_agent_data(self, agent_id):
+        """Fetch data for the given agent."""
+        url = urljoin(self.server, f"/v1/agents/data/{agent_id}")
+        try:
+            response = self.session.get(url, timeout=10)
+        except requests.exceptions.RequestException as exc:
+            logger.error(exc)
+            return {}
+        if not response:
+            logger.error(
+                "Unable to get agent data from: %s (error: %d)",
+                url,
+                response.status_code,
+            )
+            return {}
+        if response.content:
+            return response.json()
+        else:
+            return {}
```

Comment:
I think this can be simplified if we expect to return {} upon any failure.
```suggestion
        try:
            response = self.session.get(agent_data_url, timeout=30)
            response.raise_for_status()
            return response.json()
        except (requests.exceptions.RequestException, ValueError) as exc:
            logger.error("Failed to retrieve agent data: %s", exc)
            return {}
```

This suggestion comes from one of my PR where I actually want to do the same thing, you can refer to this [link](https://github.com/canonical/testflinger/blob/f715257f1d535b143aa71f043de6932e4c0aeacb/agent/src/testflinger_agent/client.py) to the check the changes I propose to this specific file. Let me know your thoughts and we can maybe agree on something that works best for both of us. 

Could you please also add typehints?